unstable test TestMultiSchemaChangeAlterIndex

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
    multi_schema_change_test.go:978: 

        	Error Trace:	ddl/multi_schema_change_test.go:978

        	            				ddl/internal/callback/callback.go:122

        	            				ddl/job_table.go:309

        	            				util/wait_group_wrapper.go:154

        	            				src/runtime/asm_amd64.s:1598

        	Error:      	Expected value not to be nil.

        	Test:       	TestMultiSchemaChangeAlterIndex

panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x499fd06]


goroutine 316875 [running]:

github.com/pingcap/tidb/ddl_test.TestMultiSchemaChangeAlterIndex.func1(0xc0648ccdc0)

	ddl/multi_schema_change_test.go:980 +0x66

github.com/pingcap/tidb/ddl/internal/callback.(*TestDDLCallback).OnJobUpdated(0xc064f6a1e0, 0xc0648ccdc0?)

	ddl/internal/callback/callback.go:122 +0x1ea

github.com/pingcap/tidb/ddl.(*ddl).delivery2worker.func1()

	ddl/job_table.go:309 +0xdef

github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run.func1()

	util/wait_group_wrapper.go:154 +0x6b

created by github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run

	util/wait_group_wrapper.go:152 +0x8d

--

Coverage runner: Not collecting coverage for failed test.

```
### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/ddl/multi_schema_change_test.go
+++ b/ddl/multi_schema_change_test.go
@@ -976,7 +976,9 @@ func TestMultiSchemaChangeAlterIndex(t *testing.T) {
 	var checked bool
 	callback := &callback.TestDDLCallback{Do: dom}
 	onJobUpdatedExportedFunc := func(job *model.Job) {
-		assert.NotNil(t, job.MultiSchemaInfo)
+		if job.MultiSchemaInfo == nil {
+			return
+		}
 		// "modify column a tinyint" in write-reorg.
 		if job.MultiSchemaInfo.SubJobs[1].SchemaState == model.StateWriteReorganization {
 			checked = true

